gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -252,31 +255,29 @@ public static List<Service> parseServices(
               transport));
     }
 
-    // Prevent codegen for mixed-in services if there are other services present, since that is an
-    // indicator that we are not generating a GAPIC client for the mixed-in service on its own.
-    Function<Service, String> serviceFullNameFn =
-        s -> String.format("%s.%s", s.protoPakkage(), s.name());
-    Set<Service> blockedCodegenMixinApis = new HashSet<>();
-    Set<Service> definedServices = new HashSet<>();
-    for (Service s : services) {
-      if (MIXIN_ALLOWLIST.contains(serviceFullNameFn.apply(s))) {
-        blockedCodegenMixinApis.add(s);
-      } else {
-        definedServices.add(s);
-      }
-    }
-    // It's very unlikely the blocklisted APIs will contain the other, or any other service.
-    boolean servicesContainBlocklistedApi =
-        !blockedCodegenMixinApis.isEmpty() && !definedServices.isEmpty();
-    // Service names that are stated in the YAML file (as mixins). Used to filter
-    // blockedCodegenMixinApis.
-    Set<String> mixedInApis =
-        !serviceYamlProtoOpt.isPresent()
-            ? Collections.emptySet()
-            : serviceYamlProtoOpt.get().getApisList().stream()
-                .filter(a -> MIXIN_ALLOWLIST.contains(a.getName()))
-                .map(a -> a.getName())
-                .collect(Collectors.toSet());
+    // Mixin services that are stated in the service yaml file.
+    List<Service> mixinServices =
+        serviceYamlProtoOpt
+            .map(
```

Comment:
This logic is getting hard to read with a lot of nested indents, can we refactor it to make it easier to read?

Reply:
Sure, will do.